.github/workflows/unit-test.yml

```
@@ -0,0 +1,79 @@
+name: Unit and linting tests
+
+on:
+  workflow_call:
+    inputs:
+      plugin:
+        type: string
+jobs:
+  unit-tests:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up Python ${{matrix.python-version}}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{matrix.python-version}}
+          cache: 'pip'
+      - name: Install dependencies
+        run: |
+          cd ${{ inputs.plugin }}
+          pip install git+https://github.com/kedro-org/kedro@main
+          pip install -r test_requirements.txt
```

```suggestion
          pip install git+https://github.com/kedro-org/kedro@main -r test_requirements.txt
```
Didn't check if this is also there is the existing CI, but we should not have multiple `pip install` commands, because they can override dependencies installed in a previous step.

---

The current CI setup does it this way as well, kedro is installed first and then the `test_requirements.txt` for the plugin being tested. I tried making this change in my forked repo but the tests start failing at the "Installing dependencies" stage for `kedro-datasets` because of dependency resolution conflict. ([See this failed run](https://github.com/ankatiyar/kedro-plugins/actions/runs/4670428836/jobs/8270142915))

---

I think this is a critical problem that needs resolving--whether it's in scope of this PR or not is a separate issue.

What this means is that, in reality, we don't have resolvable requirements, and we're only able to get to a resolvable state by overwriting some of the previously-installed requirements. Some of the stuff installed in the `pip install -r test_requirements.txt` are not actually going to be compatible with Kedro on `main`, it seems.